ops/cli.py

```diff
@@ -1,351 +1,343 @@
+        info("=>  Stopping cloud_sql_proxy")
+        shell(f"docker stop {container}")
+
+        success("Deployment is done!")
+        info(shell("kubectl get pods"), plain=True)
 
-        success("[✔] Deployment is done!")
-        shell("kubectl get pods")
         return 0
 
+    @staticmethod
+    def __set_up_helm():
+        """Add helm repo and update."""
+        info("=>  Setting up Helm charts")
+        shell("helm repo add gotc-charts https://broadinstitute.github.io/gotc-helm-repo/")
+        shell("helm repo update")
+        success("Set up Helm charts")
+
+    def __publish_docker_images(self):
+        """Publish docker build."""
+        info(f"=>  Publishing Docker images for version {self.version}")
+        for module in ["api", "ui"]:
+            shell(f"docker push broadinstitute/workflow-launcher-{module}:{self.version}")
+        success("Published Docker images")
+
+    def __helm_deploy_wfl(self, values: str):
+        """Helm deploy using gotc-charts/wfl."""
+        info(f"=>  Deploying to {self.cluster_name} in {self.cluster_namespace} namespace")
+        info("    This must run on a non-split VPN", plain=True)
+        shell(f"helm upgrade wfl-k8s gotc-charts/wfl -f {values} --install")
+        success("WFL deployed")
+
+    @staticmethod
+    def __run_liquibase_migration(db_username: str, db_password: str):
+        info("=>  Running liquibase migration")
+        db_url = "jdbc:postgresql://localhost:5432/wfl?useSSL=false"
+        changelog_dir = os.path.join(os.getcwd(), "database")
+        shell(f"docker run --rm --net=host "
+              f"-v {changelog_dir}:/liquibase/changelog liquibase/liquibase "
+              f"--url='{db_url}' --changeLogFile=/changelog/changelog.xml "
+              f"--username='{db_username}' --password='{db_password}' update")
+        success("[✔] Ran liquibase migration")
+
+
+class CLI:
+    def __init__(self):
+        """Set up the parser."""
+        parser = argparse.ArgumentParser(description="Deploy or connect to WFL infrastructure instances",
+                                         usage="%(prog)s [-h] [-d | -f] COMMAND [-e ENV] [-i INSTANCE] [...]")
+        parser.add_argument("-d", "--dry-run", action="store_true",
+                            help="Prevent COMMAND from enacting changes")
+        parser.add_argument("-f", "--force", action="store_true",
+                            help="Force COMMAND to run even if validation fails")
+        commands = parser.add_argument_group("COMMAND")
+        commands.add_argument("command", choices=["deploy", "connect", "info"],
+                              help="Deploy the local build to the instance, "
+                                   "connect to the instance's Cloud SQL, "
+                                   "or display instance config")
+        typical = parser.add_argument_group("typical arguments",
```

Comment:
Why are there so-called `typical` and `specific` arguments? I'm not sure this distinction is helpful. Either an option/switch is optional or it is not.

Reply:
Fair enough. Updated this to make environment and instance required. Can't do a whole lot about the way that argument groups work: every option that isn't in a group is an "optional argument" and comes first. Renamed this "required arguments" and the specific one "miscellaneous arguments" so they still fall in order of how most folks would execute the command

Reply:
Resolved in new version, no argument groups exist in the file.